wgengine/userspace.go

```diff
@@ -38,11 +39,13 @@ type userspaceEngine struct {
 
 	wgLock       sync.Mutex // serializes all wgdev operations
 	lastReconfig string
+	lastCfg      wgcfg.Config
 	lastRoutes   string
 
 	mu           sync.Mutex
 	peerSequence []wgcfg.Key
 	endpoints    []string
+	pingers      map[wgcfg.Key]chan struct{}
```

Comment:
I don't like the `chan struct{}` here. It's not obvious what it's used for. I'd say docs, but I'd rather we replace it with something more explicit.

Can you create a defined type for it? `CancelChan` somewhere? Then it could even have a Cancel method to do the close on itself.

Reply:
Document that mu must be held to close the channel (mu guarding map access to pingers is obvious from struct ordering convention, but the chan closure is not)